lib/redis/lock.rb

```diff
@@ -38,18 +38,19 @@ def lock(&block)
       try_until_timeout do
         expiration = generate_expiration
         # Use the expiration as the value of the lock.
-        break if redis.setnx(key, expiration)
+        break if redis.setnx(key, expiration) && Redis.next.setnx(key, expiration)
```

Comment:
perhaps it should be:
```
break if redis.setnx(key, expiration) 
   Redis.next.set(key, expiration)
```
This will make sure that OLD redis always mirrors NEW redis, regardless of what NEW redis holds.